scripts/aws-cni-support.sh

```diff
@@ -17,57 +17,59 @@
 # Set language to C to make sorting consistent among different environments.
 export LANG=C
 
-set -e
+set -euo pipefail
 LOG_DIR="/var/log/aws-routed-eni"
+mkdir -p ${LOG_DIR}
 
 # collecting L-IPAMD introspection data
-curl http://localhost:61678/v1/enis         > ${LOG_DIR}/eni.output
-curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.output
-curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.output
-curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.output
-curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.output
+curl http://localhost:61678/v1/enis         > ${LOG_DIR}/eni.out
+curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.out
+curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.out
+curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.out
+curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.out
 
-# metrics TODO not able to use LOG_DIR
-curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.output
+# metrics
+curl http://localhost:61678/metrics 2>&1 > ${LOG_DIR}/metrics.out
 
-# collecting kubelet introspection data
-curl http://localhost:10255/pods  > ${LOG_DIR}/kubelet.output
+# Collecting kubelet introspection data
+KUBECONFIG=${KUBECONFIG:-`grep kubeconfig /etc/systemd/system/kubelet.service | awk '{print $2}'`}
```

Comment:
kubeconfig file location could differ based on how the AMI is created. 

For example, if a worker node is created with eksctl, kubeconfig location `/etc/eksctl/kubeconfig.yaml` is read from eksctl-specific systemd drop-in unit file `/etc/systemd/system/kubelet.service.d/10-eksclt.al2.conf` and fail with below error:

`error: unable to read certificate-authority /etc/kubernetes/pki/ca.crt for kubernetes due to open /etc/kubernetes/pki/ca.crt: no such file or directory`

Reply:
Thanks for checking, will have to improve this handling a bit then.

Reply:
Added specific logic for eksctl